src/posix/platform/spi_interface.cpp

```diff
@@ -751,68 +760,48 @@ void SpiInterface::Process(const RadioProcessContext &aContext)
 
 otError SpiInterface::WaitForFrame(uint64_t aTimeoutUs)
 {
-    otError        error      = OT_ERROR_NONE;
-    struct timeval spiTimeout = {kSecPerDay, 0};
-    struct timeval timeout;
-    fd_set         readFdSet;
-    int            ret;
-    bool           isDataReady = false;
+    otError  error = OT_ERROR_NONE;
+    uint64_t now   = otPlatTimeGet();
+    uint64_t end   = now + aTimeoutUs;
 
-    timeout.tv_sec  = static_cast<time_t>(aTimeoutUs / US_PER_S);
-    timeout.tv_usec = static_cast<suseconds_t>(aTimeoutUs % US_PER_S);
+    mDidRxFrame = false;
 
-    FD_ZERO(&readFdSet);
-
-    if (mIntGpioValueFd >= 0)
+    while (now < end)
```

Comment:
I do not think we need the `!mDidRxFrame` check here.
- When we enter the `while()` loop for the first time, we know for sure `mDidRxFrame` is `false`, since we set it to `false` at the top of this method.
- The only option for `mDidRxFrame` to be set to `true` is from the `Process()` call below, where we immediately check it `ExitNow()` if it gets `true`.

Reply:
I meant that we can remove check and ExitNow to make the code more concise. 
But fine to keep it as is.

Reply:
I see. I think the code now reads better. It is clear we `ExitNow()` on `mDidRxFrame` getting `true`, and if we come out of loop, we have error `TIMEOUT` (no need to check again outside the loop to decide on `error`).